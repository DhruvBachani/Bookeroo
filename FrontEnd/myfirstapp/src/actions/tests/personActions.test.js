import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore([]);

const createPerson = () => ({
    type: 'GET_ERRORS'
})

const getPersons = () => ({
    type: 'GET_PERSONS'
});

const getPerson = () => ({
    type: 'GET_PERSON'
});

describe('personActions', () => {
    it('creates GET_ERRORS when a person cannot be created', () => {
        store.dispatch(createPerson());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_PERSONS when it is called successfully', () => {
        store.dispatch(getPersons());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates GET_PERSON when it is called successfully', () => {
        store.dispatch(getPerson());
        expect(store.getActions()).toMatchSnapshot();
    })
})