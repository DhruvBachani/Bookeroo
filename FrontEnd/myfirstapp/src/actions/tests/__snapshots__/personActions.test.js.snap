// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`personActions creates GET_PERSON when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_PERSONS",
  },
  Object {
    "type": "GET_PERSON",
  },
]
`;

exports[`personActions creates GET_PERSONS when it is called successfully 1`] = `
Array [
  Object {
    "type": "GET_PERSONS",
  },
]
`;
